activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -415,12 +405,6 @@ def write_serialized_entry(key, payload, raw: false, unless_exist: false, expire
           end
         end
 
-        def write_key_expiry(client, key, options)
-          if options[:expires_in] && client.ttl(key).negative?
-            client.expire key, options[:expires_in].to_i
-          end
```

Comment:
I'm actually curious of why this behavior, if I'm understanding correctly:

if we pass an expiry, we only set it if the counter didn't already have one? What's the point of this? Not to update the expiry on increment? seems wasteful and I'd actually expect the expiry to be refreshed when I increment / decrement the counter. 

I'll do some git spelunking and check other implementation to see why this was done this way.

Reply:
Ok, so it was added in https://github.com/rails/rails/pull/33254 to replicate the memcached behavior. 

IMO we should instead change `MemCacheStore` to bump the expiry after an increment.

Reply:
Or even just accept the disrependency here. Not 100% sure. IMO this is a Memcached limitation, and while I understand that most AS::Cache backends try to behave as closely to Memcached as possible I don't think it's a good call here.

Reply:
> I'd actually expect the expiry to be refreshed when I increment / decrement the counter.

This is also what I was thinking. But the current behavior kinda makes sense. For example, when the counter's cache does not include the time (like `user-1`) and we want to keep the value for some period of time (like 1 minute). If we will update the expiration time every time we increment it, this counter can be kept longer then needed and have not the expected value.

Reply:
I really don't see how that's useful. TTL on a counter is generally for garbage collection of no longer used counters.

Also if you overwrite an existing key, you refresh the TTL, but somehow incrementing/decrementing is not considered like an "overwrite"? Seems unexpected to me.

On another note, I checked and Dalli doesn't have proper pipelining support, so issuing a `touch` after the `incr` would impact performance, so maybe we should just document that limitation in `MemCacheStore` and let `RedisCacheStore` do the sensible thing.

Reply:
> I really don't see how that's useful. TTL on a counter is generally for garbage collection of no longer used counters.

Contrived example: we want to keep track of how many requests user made per minute. We do `incr("user-$id:requests", expires_in: 1.minute)`, and `get("user-$id:requests")` to get the counter. So each minute this counter will be expired and we start incrementing from 0 again. If we will update ttl each time, this counter could not be reset each minute.

> so maybe we should just document that limitation in MemCacheStore and let RedisCacheStore do the sensible thing.

Should this better done as a follow up, if we will agree on this?

Reply:
Seems very contrived indeed, and probably better handled with a namespaced key...

FYI your PR prompted me to do a bit of a cleanup in `RedisCacheStore` https://github.com/rails/rails/pull/45721, one of the change is that rather than check for `mset` it now check whether you can use a pipeline. The impact on your PR is that rather than to use a Redis 7.0 feature, you can just pipeline. So let's wait a tiny bit on your PR, I'll ping you when I'm done and you can update your PR.